docs/src/tutorials/CountAndCache.md

```diff
@@ -0,0 +1,164 @@
+# How to Cache function calls and Get Started: Optimize!
+Ronny Bergmann
+
+In this tutorial, we want to investigate the caching and counting (i.e. statistics) features
+of [Manopt.jl](https://manoptjl.org). We will reuse the optimization tasks from the
+introductionary tutorial [Get Started: Optimize!](https://manoptjl.org/stable/tutorials/Optimize!.html).
+
+## Introduction
+
+THere are surely many ways to keep track for example of how often the cost function is called,
+for example with a [functor](https://docs.julialang.org/en/v1/manual/methods/#Function-like-objects), as we used in an example in [How to Record Data](https://manoptjl.org/stable/tutorials/HowtoRecord.html)
+
+``` julia
+mutable struct MyCost{I<:Integer}
+    count::I
+end
+MyCost() = MyCost{Int64}(0)
+function (c::MyCost)(M, x)
+    c.count += 1
+    # [ .. Actual implementation of the cost here ]
+end
+```
+
+though this still leaves a bit of work to the user, especially for trakcing more than just the cost.
+
+Similarly caching of function calls can for example be added using [Memoize.jl](https://github.com/JuliaCollections/Memoize.jl) by the user.
+The approach in the solvers of [Manopt.jl](https://manoptjl.org) aims to simplify adding
+both these capabilities on the level of calling a solver.
```

Comment:
```suggestion
When the a function like objective or gradient is expensive to compute, it may make sense to cache its results.
Manopt.jl tries to minimize the number of repeated calls but sometimes they are necessary and harmless when the function is cheap to compute.
Caching of expensive function calls can for example be added using [Memoize.jl](https://github.com/JuliaCollections/Memoize.jl) by the user.
The approach in the solvers of [Manopt.jl](https://manoptjl.org) aims to simplify adding
both these capabilities on the level of calling a solver.
```
I think a bit more details as to why a user would want caching is a good idea.